docs/worldgen/biomemodifier.md

```diff
@@ -0,0 +1,648 @@
+        "maxCount": 10
+      }
+    ]
+}
+```
+
+### Remove Spawns
+
+This Biome Modifier type removes mob spawns from biomes. The modifier takes in the `Biome` id or tag of the biomes the spawning information are removed from and the `EntityType` id or tag of the mobs to remove.
+
+```json5
+{
+    "type": "forge:remove_spawns",
+
+    // Can either be an id "minecraft:plains"
+    // List of ids ["minecraft:plains", "minecraft:badlands", ...]
+    // Or a tag "#c:is_overworld"
+    "biomes": "#namespace:biome_tag",
+
+    // Can either be an id "minecraft:ghast"
+    // List of ids ["minecraft:ghast", "minecraft:skeleton", ...]
+    // Or a tag "#minecraft:skeletons"
+    "entity_types": "#namespace:entitytype_tag"
+}
+```
+
+### Add Spawn Costs
+
+Allows for adding new Spawn Costs to biomes. Spawn Costs are a newer way of making mobs spawn spread out in a biome to reduce clustering. The modifier takes in the `Biome` id or tag of the biomes the spawn costs are added to, the `EntityType` id or tag of the mobs to add spawn costs for, and the `MobSpawnSettings.MobSpawnCost` of the mob. The `MobSpawnCost` contains the energy budget, which indicates the maximum number of entities that can spawn in a location based upon the charge provided for each entity spawned.
+
+```json5
+{
+    "type": "neoforge:add_spawn_costs",
+    // Can either be an id "minecraft:plains"
+    // List of ids ["minecraft:plains", "minecraft:badlands", ...]
+    // Or a tag "#c:is_overworld"
+    "biomes": "#c:is_overworld",
+    // Can either be an id "minecraft:ghast"
+    // List of ids ["minecraft:ghast", "minecraft:skeleton", ...]
+    // Or a tag "#minecraft:skeletons"
+    "entity_types": "#minecraft:skeletons",
+    "spawn_cost": {
+        // The energy budget
+        "energy_budget": 1.0,
+        // The amount of charge each entity takes up from the budget
+        "charge": 0.1
+    }
+}
+```
+
+### Remove Spawn Costs
+
+Allows for removing a Spawn Cost from a biome. Spawn Costs are a newer way of making mobs spawn spread out in a biome to reduce clustering. The modifier takes in the `Biome` id or tag of the biomes the spawn costs are removed from and the `EntityType` id or tag of the mobs to remove the spawn cost for.
+
+```json5
+{
+    "type": "neoforge:remove_spawn_costs",
+    // Can either be an id "minecraft:plains"
+    // List of ids ["minecraft:plains", "minecraft:badlands", ...]
+    // Or a tag "#c:is_overworld"
+    "biomes": "#c:is_overworld",
+    // Can either be an id "minecraft:ghast"
+    // List of ids ["minecraft:ghast", "minecraft:skeleton", ...]
+    // Or a tag "#minecraft:skeletons"
+    "entity_types": "#minecraft:skeletons"
+}
+```
+
+### Add Legacy Carvers
+
+This Biome Modifier type allows adding old caves and ravines to biomes. This CANNOT add Noise Caves to biomes because Noise Caves are baked into the dimension's Noise Setting system and not actually tied to biomes. The legacy carvers are specifically ravines and old caves.
```

Comment:
If you don't know the kinds of caves there are, you probably should not be touching this modifier anyway. Though I am a bit surprised you don't know what a ravine is as that's pretty iconic of minecraft lol. I can describe the old caves as the pre-caves and cliffs cave styles. But I can't really explain a ravine. Or noise caves. That's a whole wiki 
https://minecraft.wiki/w/Cave

I guess I can say Carver Cave instead.

Old Caves: 
![image](https://github.com/user-attachments/assets/e0dde3e5-9b37-4284-8bd5-f01b7dcb9351)

Ravines:
![ravine-finder_400](https://github.com/user-attachments/assets/3f3f24cc-407c-46ca-b284-a22144b01021)

Noise Caves:
![image](https://github.com/user-attachments/assets/c2c45420-9112-4596-910f-babe4e0c7673)
![image](https://github.com/user-attachments/assets/2610d8f4-117a-4747-a4b8-caa314ff3f01)

Reply:
I phrased my statement wrong. I know what the differences are, but to me, the words 'old cave' and noise cave didn't mean much. What I know them as is the cave and canyon carvers, whereas the noise cave is just the settings of the density functions on the noise router of the chunk generator. If these are names people in the community are familiar with, then ignore me. I just didn't recognize them since I never refer to them that way.